core/coreunix/add.go

```diff
@@ -316,6 +320,13 @@ func (adder *Adder) AddAllAndPin(file files.Node) (ipld.Node, error) {
 		return nil, err
 	}
 
+	if asyncDagService, ok := adder.dagService.(syncer); ok {
+		err = asyncDagService.Sync()
+		if err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
A little unfortunate that in the common case we'll end up syncing the same key twice since right after this we call `PinRoot` which calls `pin.Flush()`

Reply:
Yeah... but hopefully, this will basically be a no-op.